source/routing/get_stop_times.cpp

```diff
@@ -181,12 +182,12 @@ get_all_calendar_stop_times(const routing::JourneyPattern& jp,
             //Note: end can be lower than start, so we have to cycle through the day
             const auto freq_vj = static_cast<const type::FrequencyVehicleJourney*>(vj);
             bool is_looping = (freq_vj->start_time > freq_vj->end_time);
-            auto stop_loop = [freq_vj, is_looping](u_int32_t t) {
+            auto stop_loop = [freq_vj, is_looping, st](u_int32_t t) {
                 if (! is_looping)
-                    return t <= freq_vj->end_time;
-                return t > freq_vj->end_time;
+                    return t <= freq_vj->end_time + st.departure_time;
+                return t > freq_vj->end_time + st.departure_time;
             };
-            for (auto time = freq_vj->start_time; stop_loop(time); time += freq_vj->headway_secs) {
+            for (auto time = freq_vj->start_time + st.departure_time; stop_loop(time); time += freq_vj->headway_secs) {
```

Comment:
I saw something wrong while testing frequencies with boarding / alighting (since the start_time could be the first boarding_time).
But I think if we currently have the arrival_time of the first stop_time smaller than its departure_time it would break too, because we assume the start_time is the departure_time but it's based on the arrival_time.